Mathlib/Analysis/Calculus/FDeriv/Add.lean

```diff
@@ -531,16 +531,85 @@ theorem DifferentiableAt.sub (hf : DifferentiableAt 𝕜 f x) (hg : Differentiab
   (hf.hasFDerivAt.sub hg.hasFDerivAt).differentiableAt
 #align differentiable_at.sub DifferentiableAt.sub
 
+@[simp]
+lemma DifferentiableAt.add_iff_left (hg : DifferentiableAt 𝕜 g x) :
+    DifferentiableAt 𝕜 (fun y => f y + g y) x ↔ DifferentiableAt 𝕜 f x := by
+  refine ⟨fun h ↦ ?_, fun hf ↦ hf.add hg⟩
+  simpa only [add_sub_cancel_right] using h.sub hg
+
+@[simp]
+lemma DifferentiableAt.add_iff_right (hg : DifferentiableAt 𝕜 f x) :
+    DifferentiableAt 𝕜 (fun y => f y + g y) x ↔ DifferentiableAt 𝕜 g x := by
+  simp only [add_comm (f _), hg.add_iff_left]
+
+@[simp]
+lemma DifferentiableAt.sub_iff_left (hg : DifferentiableAt 𝕜 g x) :
+    DifferentiableAt 𝕜 (fun y => f y - g y) x ↔ DifferentiableAt 𝕜 f x := by
+  refine ⟨fun h ↦ ?_, fun hf ↦ hf.sub hg⟩
+  simpa only [sub_add_cancel] using h.add hg
```

Comment:
```suggestion
  simp [sub_eq_add_neg, hg]
```
should work, hopefully. If it doesn't, please make it work by adding `DifferentiableAt 𝕜 (fun y => -f y) ↔ DifferentiableAt 𝕜 f x` as a simp lemma

Reply:
Does work. Replaced by `simp only` instead because I'm guessing it's faster and more stable to future changes.

Reply:
Faster, a bit. More stable to future changes, not necessarily since `simp only` requires you to name lemmas which might be renamed in the future

Reply:
> `simp only` requires you to name lemmas which might be renamed in the future
I would think renaming lemmas is much easier than figuring out which new `simp` lemma is now preventing a proof from working and why.

Reply:
In my experience, not necessarily, no. Further, renamings happen much more often than terminal simp calls not working anymore